ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -275,12 +280,17 @@ void RSkComponentTextInput::processEventKey (rnsKey eventKeyType,bool* stopPropa
     //is always 0 & selectionRange.location always end 
     textInputMetrics.selectionRange.location = cursor_.end ;
     textInputMetrics.selectionRange.length = 0;
+    int newLength = textString.length();
     if (updateString) {
       if (displayString_ != textString){
-        displayString_ = textString;
-        cursor_.end = textString.length();
-        flushLayer();
+        if ((newLength > maxLength_) && (newLength > prevLength)) {
+          std::copy_n(textString.begin(), prevLength, displayString_.begin());
+        } else {
+          displayString_ = textString;
+        }
       }
+      cursor_.end = textString.length();
```

Comment:
this need to be displayString_.length() ?
case : 
if textString.length is greater than maxLength, we are copying only prevLength in displayString